### DISCUSSION TITLE
Support for JSONSchema '"type" as "array", with a custom format, and the data for this field is coming from async service.

### DISCUSSION THREAD

**Discussion by aghosh-evertz** [json-schema-org/community#448] [Q&A]
Hi All,
I have tried to go through the doc to find, if anything like this is supported:

"Support for JSONSchema '"type" as "array", when we have a custom format added to it through code, and the field values are recieved through an async service."

Right now, if I have a schema like this:

JSONSchema:

```json
{
  "$schema": "http://json-schema.org/draft-07/schema",
  "type": "object",
  "title": "Incident Filter",
  "definitions": {
    "channelIds": {
      "type": "array",
      "title": "Channel names",
      "description": "Channels which will be monitored for incidents"
    }
  },
  "properties": {
    "channelIds": {
      "$id": "#/properties/channelIds",
      "$ref": "#/definitions/channelIds"
    }
  }
}
```

The format is getting added through a separate patch file like this:

```
import { JSONSchema7 } from 'json-schema';

import { JsonSchemaPatches } from '@evutils/formly-utils';

export const updateSchemaFields: JsonSchemaPatches = {
  channelIds: updateChannelFieldConfig,
};

function updateChannelFieldConfig(json: JSONSchema7): JSONSchema7 {
  return {
    ...json,
    format: 'channel-custom-format',
  };
}
```

The values for this channel-field is coming from the service, which is later added, when available (async).

It doesn't work and throws random error. How are we supposed to work with a schema field having 'format',whose type is "array", but values are not instantly available??

**Comment by Relequestual** [comment]
It sounds like you have an async/await code problem here.
You should only try to validate your JSON data when you have it all.
There's nothing JSON Schema specific about this. Sorry.
